### ISSUE TITLE
Read Only Permissions include Modify capabilities

### ISSUE THREAD

**Issue by 808d1** [microsoft/Microsoft365DSC#3404]
I have found that I am unable to export configurations for Office 365 and Teams without including ReadWrite permissions.  Can someone confirm this to be a requirement to only export settings?

- **For Office 365** I needed to add Graph API User.ReadWrite.All after receiving permission errors from "Get-MgOrganizationSettingPersonInsight".  Using Find-MgGraphCommand, I was able to determine the ReadWrite permssion was needed.
- **For Teams** I needed to add Graph API "Group.ReadWrite.All".  This seemed to be confirmed by the following article in reference to the Non *-CS cmdlets:
https://learn.microsoft.com/en-us/microsoftteams/teams-powershell-application-authentication#setup-application-based-authentication

For reference, here are my current Graph API settings:
1. AdministrativeUnit.Read.All
2. Application.Read.All
3. ChannelSettings.Read.All
4. Directory.Read.All
5. Domain.Read.All
6. Group.Read.All
7. Group.ReadWrite.All
8. Organization.Read.All
9. User.Read.All
10. User.ReadWrite.All

**Comment by NikCharlebois** [comment]
Are you authenticating using Credentials or Service Principal?

**Comment by NikCharlebois** [comment]
Please take a look at:
https://github.com/microsoft/Microsoft365DSC/issues/3382

**Comment by 808d1** [comment]
> Are you authenticating using Credentials or Service Principal?

I am using a Service Principal with certificate thumbprint

**Comment by 808d1** [comment]
> 



> 

Based on #3382, am I correct to assume this is a known limitation/issue with Service Principals at this time?

**Comment by NikCharlebois** [comment]
That is correct. Unfortunately that resource doesn't yet support app-only authentication. We will review once support gets added in the future.

**Comment by ykuijs** [comment]
Updating required permissions in documentation via PR #3636
